lib/Index/Index.cpp

```diff
@@ -153,6 +154,27 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
 
   bool IsModuleFile = false;
   bool isSystemModule = false;
+
+  struct ContainerStackEntry {
+    Decl *const visitedDecl;
+    // These vector capacities represent the number of declarations that may
+    // be associated with the key type. In most cases there will be only one; a
+    // function or single named pattern binding. However in the case of an
+    // expanded tuple pattern binding, there will be multiple.
+    SmallVector<const Decl *, 6> activeDecls{};
+    llvm::DenseMap<
+        const void *,
+        std::pair<bool, llvm::PointerUnion<const Decl *, const Pattern *>>>
```

Comment:
Is the `bool` needed here? It seems like a `nullptr` represents the same thing - all `{false, *}` have a value and the only `true` is `{true, nullptr}`